ethereumetl/deduplication/deduplicate.py

```
@@ -0,0 +1,67 @@
+import os
+import logging
+import asyncio
+from datetime import datetime
+from ethereumetl.constants import constants
+
+def deduplicate_records(records, ts_key, db):
+    ch_fallback_days = int(os.environ.get('CLICKHOUSE_FALLBACK_DAYS', constants.CLICKHOUSE_FALLBACK_DAYS))
+    
+    if records == None or len(records) == 0:
+        return records
+    
+    min_ts = get_minimum_ts(records, ts_key)
+    if is_ts_older(min_ts, ch_fallback_days):
+        records = asyncio.run(filter_records(records, min_ts, db))
+    return records
+
+def is_ts_older(ts, days):
+    difference = datetime.utcnow() - datetime.utcfromtimestamp(ts)
+    return difference.days > days    
+
+async def filter_records(items, min_ts_epoch, db):
+    if items == None or len(items) == 0:
+        return items
+    
+    message_type = items[0].get('type')
+    skip_message_types = os.environ.get('CLICKHOUSE_SKIP_FOR_MESSAGE_TYPES')
+    
+    if skip_message_types != None and (message_type in skip_message_types.split(',')):
+        logging.info(f'Ignoring check for deduplication for type {message_type} as it is ignored')
+        return items
+    
+    table_name = get_table_name(message_type)
+    ts_column_name = get_table_ts_column_name(message_type)
+    if table_name == None:
+        logging.warn(f'Ignoring check for deduplication for type {message_type} as table not found')
```

It should be panic instead of warning. service should stop execution.